components/gitpod-protocol/src/messaging/node/connection.ts

```diff
@@ -38,3 +44,22 @@ export function toIWebSocket(ws: ws) {
         },
     };
 }
+
+// copied from /node_modules/vscode-ws-jsonrpc/lib/socket/connection.js
+export function createWebSocketConnection(socket: IWebSocket, logger: Logger) {
+    const messageReader = new SafeWebSocketMessageReader(socket);
+    const messageWriter = new WebSocketMessageWriter(socket);
+    const connection = createMessageConnection(messageReader, messageWriter, logger);
+    connection.onClose(() => connection.dispose());
+    return connection;
+}
+
+class SafeWebSocketMessageReader extends WebSocketMessageReader {
+    protected readMessage(message: any): void {
+        try {
+            super.readMessage(message);
+        } catch (error) {
+            console.error("Failed to decode JSON-RPC meessage.", error);
```

Comment:
```suggestion
            log.error("Failed to decode JSON-RPC message.", error);
```

Reply:
I'm unsure about the exact error message, but IMO it should not be error.

@akosyakov Didn't we have some docs on "how to do logging"? Can't find it in notion atm :thinking:

Reply:
I don't believe so, only on LOG reduction policies.

Reply:
`log` it is then.

Changed to `debug`, but "it should not be error" lacks of reasoning. It _is_ an error.

Reply:
I wonder should we rather distinguish here between json parsing errors in other?

Reply:
> but "it should not be error" lacks of reasoning. It is an error.

Fair point. Would be great if we could gather more examples, discuss and document our decision in one place. :+1: